src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -453,6 +458,34 @@ public static ItemStack setStack(@NotNull final IInventory inventory, @Nullable
     }
 
     /**
+     * {@link #setStack(IInventory, ItemStack)}.
+     * Tries to put an itemStack into Inventory, unlike setStack, allow to use a ItemStack bigger than the maximum stack size allowed for the item
+     *
+     * @param inventory the inventory to set the stack in.
+     * @param stack     Item stack with items to be transferred, the stack can be bigger than allowed
+     * @return returns null if successful, or stack of remaining items, BE AWARE that the remaining stack can be bigger than the maximum stack size
+     */
+    @Nullable
+    public static ItemStack setOverSizedStack(@NotNull final IInventory inventory, @Nullable final ItemStack stack)
+    {
+        int stackSize = stack.stackSize;
+        while (stackSize > 0)
+        {
+            final int itemCount = Math.min(stackSize, stack.getMaxStackSize());
+            final ItemStack items = new ItemStack(stack.getItem(), itemCount, stack.getItemDamage());
+            stackSize-=itemCount;
+            final ItemStack remainingItems = setStack(inventory,items);
+            stackSize+=remainingItems.stackSize;
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'remainingItems' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2259)